pylti1p3/message_launch.py

```diff
@@ -313,22 +457,29 @@ def validate_nonce(self):
         return self
 
     def validate_registration(self):
+        # type: (T_SELF) -> T_SELF
         iss = self._get_iss()
 
+        # Check client id
+        aud = self._get_jwt_body().get('aud')
+        client_id = aud[0] if isinstance(aud, list) else aud
+
         # Find registration
-        self._registration = self._tool_config.find_registration_by_issuer(iss)
+        self._registration = self._tool_config.find_registration_by_issuer_and_client_id(iss, client_id)
```

Comment:
As we discussed before - let's reuse `find_registration_by_issuer` with additional param